automation/src/test/scala/org/broadinstitute/dsde/workbench/leonardo/apps/AppCreationSpec.scala

```diff
@@ -83,11 +85,17 @@ class KubernetesSpec extends GPAllocFixtureSpec with LeonardoTestUtils with GPAl
             app1DeletedDoneCheckable
           ).compile.lastOrError
 
-          _ <- loggerIO.info(s"app1 delete result: $monitorApp1DeletionResult")
+          _ <- loggerIO.info(
+            s"AppCreationSpec: app ${googleProject.value}/${appName.value} delete result: $monitorApp1DeletionResult"
+          )
+
+          _ = app1DeletedDoneCheckable.isDone(monitorApp1DeletionResult) shouldBe true
 
-          _ <- testTimer.sleep(480 seconds)
+          // TODO investigate why this is necessary - in theory the second app should be able
+          // to be created after the first is deleted.
```

Comment:
Should we add a ticket for ^^? If it's a real bug/limitation, it would be good to fix/know.

Reply:
Agreed - I'm actually trying to investigate it now, I feel like there is some subtle behavior these tests are exposing that I'm missing. (I don't think it's related to concurrent nodepool operations)